includes/wcpdf-functions.php

```diff
@@ -320,11 +320,24 @@ function wcpdf_deprecated_function( $function, $version, $replacement = null ) {
 
 /**
  * Logger function to capture errors thrown by this plugin, uses the WC Logger when possible (WC3.0+)
+ * 
+ * @param string         $message
+ * @param string         $level
+ * @param \Exception     $e
+ * @param Order_Document $document
+ * 
+ * @return void
  */
-function wcpdf_log_error( $message, $level = 'error', $e = null ) {
+function wcpdf_log_error( string $message, string $level = 'error', \Exception $e = null, WPO\WC\PDF_Invoices\Documents\Order_Document $document = null ): void {
 	if ( function_exists( 'wc_get_logger' ) ) {
 		$logger  = wc_get_logger();
-		$context = array( 'source' => 'wpo-wcpdf' );
+		$source = 'wpo-wcpdf';
+
+		if($document) {
+			$source.= '-' . $document->get_type() . '-' . $document->get_number()->get_formatted();
```

Comment:
This will create a log for each document number, I'm not sure if this is the best solution. I think having one for each document type would be preferable.

Reply:
I agree that this will produce lots of different logs, and I agree that we could have one log for each document type. I did it this way because I thought that `separate logs for each document` in issue description means `1 document = 1 log`. But of course, `1 document type = 1 log` also makes sense.
@MohamadNateqi do you agree with `1 document type = 1 log` approach?

Reply:
@BrunoPavlinic98 I also agree `1 document type = 1 log` would be better.